potassco/reify.h

```diff
@@ -0,0 +1,150 @@
+// {{{ MIT License
+
+// Copyright 2017 Roland Kaminski
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+
+// }}}
+
+#pragma once
+
+#include <cstdint>
+#include <potassco/aspif.h>
+#include <potassco/graph.h>
+#include <potassco/reify_utils.h>
+
+#include <unordered_map>
+#include <vector>
+
+namespace Potassco {
+//! Writes a program in reified facts format to the given output stream.
+/*!
+ * \ingroup WriteType
+ */
+class Reifier : public Potassco::AbstractProgram {
```

Comment:
General question also @rkaminsk: Is order really relevant for all tuples?
I.e. do we really need to create separate atom/literal tuples if they only differ in their order or could we also have "canonical" tuples instead?
Example:
```cpp
{a;b}.
{b;a}.
```
This will currently produce two atom `atom_tuple`'s but it could probably also only produce one.